trino-connector/src/test/java/com/datastrato/gravitino/trino/connector/TestGravitinoConnector.java

```diff
@@ -70,10 +75,14 @@ protected QueryRunner createQueryRunner() throws Exception {
         properties.put("gravitino.uri", "http://127.0.0.1:8090");
         queryRunner.createCatalog("test1", "gravitino", properties);
       }
-      server.setCatalogConnectorManager(GravitinoPlugin.catalogConnectorManager);
+
+      CatalogConnectorManager catalogConnectorManager =
+          gravitinoPlugin.getCatalogConnectorManager();
```

Comment:
I think there is a real Gravitino client in the CatalogConnectorManager, which may be used to load catalogs. You can pass the mocked Gravitino client to the TestGravitinoFactory to initialize the CatalogConnectorManager.

Reply:
The code snippet:

`
76 queryRunner.createCatalog("test1", "gravitino", properties);
`
calls the GravitinoConnector.create() function, which instantiates a CatalogConnectorManager. When start() is invoked on the CatalogConnectorManager, it creates an instance of GravitinoClient. This instance is a real GravitinoClient object.

Later, the line:

`
81 catalogConnectorManager.setGravitinoClient(gravitinoClient);
`
replaces the instance with a mocked GravitinoClient.

At this point, the real GravitinoClient may still be performing background tasks.

Reply:
Done